### PR TITLE
334 alien partitions api

### LINKED ISSUE #334: Add API method to create new partition for aliens
This is required for alien recovery tool

### PR DESCRIPTION
**Pull request by idruzhitskiy**
closes #334

### CODE REVIEW

File: bob/src/api/http.rs [hunk 1]
```diff
@@ -568,6 +584,16 @@ async fn alien(_bob: &State<BobServer>) -> &'static str {
     "alien"
 }
 
+#[post("/alien/sync")]
+async fn sync_alien_data(bob: &State<BobServer>) -> Result<StatusExt, StatusExt> {
+    let backend = bob.grinder().backend().inner();
+    let (_, adc) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+    adc.detach_all().await?;
```
Comment (ikopylov) [inline bob/src/api/http.rs:593]: Is this a correct command? It looks like it is detaching (not syncing) all blobs in Bob (not just aliens)

File: bob-common/src/configs/node.rs [hunk 2]
```diff
@@ -476,9 +475,9 @@ pub struct Node {
     metrics: Option<MetricsConfig>,
 
     #[serde(skip)]
-    bind_ref: RefCell<String>,
+    bind_ref: Arc<Mutex<String>>,
```
Comment (ikopylov) [inline bob-common/src/configs/node.rs:478]: Why this was changed?

File: bob-backend/src/pearl/group.rs [hunk 3]
```diff
@@ -330,9 +350,16 @@ impl Group {
...
 
-    pub fn create_pearl_by_timestamp(&self, time: u64) -> Holder {
-        let start_timestamp =
-            Stuff::get_start_timestamp_by_timestamp(self.settings.timestamp_period(), time);
+    pub fn create_pearl_by_timestamp(
+        &self,
+        time: u64,
+        timestamp_config: &StartTimestampConfig,
+    ) -> Holder {
+        let start_timestamp = Stuff::get_start_timestamp_by_timestamp(
```
Comment (ikopylov) [inline bob-backend/src/pearl/group.rs:358]: @pyakushin please, check carefully that the new logic for setting timestamps does not break anything

**Comment by ikopylov** [comment]
@idruzhitskiy this request contains changes in the core logic. Please, check again that everything is correct

File: bob-common/src/configs/node.rs (hunk 2 above)
Comment (idruzhitskiy) [reply bob-common/src/configs/node.rs:478]: This structure is now shared between multiple threads, so we need to use thread-safe abstractions

File: bob/src/api/http.rs (hunk 1 above)
Comment (idruzhitskiy) [reply bob/src/api/http.rs:593]: `adc` is `AliensDiskController`, so we are detaching (making inactive) all alien blobs

Comment (ikopylov) [reply bob/src/api/http.rs:593]: Can you make variable name more intuitive? E.g. `aliensDiskController`

Comment (idruzhitskiy) [reply bob/src/api/http.rs]: Done, changed all occurrences of `adc` in `http.rs` to `alien_disk_controller`

File: bob-backend/src/pearl/group.rs (hunk 3 above)
Comment (pyakushin) [reply bob-backend/src/pearl/group.rs:359]: Checked, everything is ok, behavior changed only for alien puts.

File: bob/src/api/http.rs [hunk 4]
```diff
@@ -313,18 +320,27 @@ async fn distribution_function(bob: &State<BobServer>) -> Json<DistrFunc> {
     })
 }
 
+#[get("/configuration")]
+async fn get_node_configuration(bob: &State<BobServer>) -> Json<NodeConfiguration> {
+    let grinder = bob.grinder();
+    let config = grinder.node_config();
+    Json(NodeConfiguration {
+        root_dir: config.pearl().blob_file_name_prefix().to_owned(),
```
Comment (pyakushin) [inline bob/src/api/http.rs:328]: It's not a root_dir

File: bob/src/api/http.rs [hunk 5]
```diff
@@ -568,13 +584,23 @@ async fn alien(_bob: &State<BobServer>) -> &'static str {
     "alien"
 }
 
+#[post("/alien/sync")]
+async fn sync_alien_data(bob: &State<BobServer>) -> Result<StatusExt, StatusExt> {
```
Comment (pyakushin) [inline bob/src/api/http.rs:588]: Ambiguous route an fn names. It detaches all alien holders, why don you named it like this?

**Review by pyakushin — CHANGES_REQUESTED** [review]

File: bob/src/api/http.rs (hunk 5 above)
Comment (idruzhitskiy) [reply bob/src/api/http.rs:588]: From the user point of view it just syncs alien data with disk. I can change it to `alien/detach`

**Pull request merged**